src/mixer/basetrackplayer.h

```diff
@@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
     ControlObjectSlave* m_pPreGain;
     ControlObjectSlave* m_pSpeed;
     ControlObjectSlave* m_pPitchAdjust;
+    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
+    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
```

Comment:
> The parent pointer documents the ownership as well, as all other QObject based classes.

This is true in that all code is its own documentation. But it's not very good documentation since it requires you to read the entire implementation to understand it. This documents the fact in the type of the variable which is much more clear.

> A scoped pointer here will confuse the reader 

A scoped pointer is a standard engineering practice -- there's really nothing confusing about it. 

> requires extra CPU.

Are you serious about this?